.gitignore

```diff
@@ -1,5 +1,250 @@
-build.sh
+# Created by https://www.toptal.com/developers/gitignore/api/cmake,python,c++,linux
+# Edit at https://www.toptal.com/developers/gitignore?templates=cmake,python,c++,linux
+
+### C++ ###
+# Prerequisites
+*.d
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+*.obj
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+*.dll
+
+# Fortran module files
+*.mod
+*.smod
+
+# Compiled Static libraries
+*.lai
+*.la
+*.a
+*.lib
+
+# Executables
+*.exe
+*.out
+*.app
+
+### CMake ###
+CMakeLists.txt.user
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Testing
+Makefile
```

Comment:
Wait, are we ignoring Makefiles?

Reply:
Aha, so apparently CMake people want us to ignore the competition! My guess is that CMake might generate Makefiles? But I agree, it our context it makes no sense

Reply:
Yeah, problem is that I have a good old Makefile at the top level.

I get the idea, but imo Makefiles generated by cmake should ideally always in a build directory, that we completely 'ignore'

Reply:
cool. I let you push the green button if you want this in :-)